File: src/no/omb/bigboat/data/RaceData.java
package no.omb.bigboat.data;

public class RaceData {

	public static final RaceData[] races = {
		new RaceData("Bjørvikasprinten", DataFileType.M2S_CSV),
		new RaceData("Isbrytern", DataFileType.M2S_CSV),
		new RaceData("Bundefjorden Vårcup", DataFileType.M2S_CSV),
		new RaceData("Oslofjorden Rundt", DataFileType.SEILMAG_CSV3),
//		new RaceData("Håøya Rundt", DataFileType.SEILMAG_CSV3),
//		new RaceData("SætreKruset", DataFileType.SEILMAG_CSV3),
//		new RaceData("Nesodden Høstcup", DataFileType.SEILMAG_CSV2),
//		new RaceData("UllernCupen Familie SH regatta 1", DataFileType.ULLERN_CSV),
//		new RaceData("UllernCupen Familie SH regatta 2", DataFileType.ULLERN_CSV),
//		new RaceData("Hurum Golden Cup", DataFileType.SEILMAG_CSV),
	};

	public enum DataFileType {
		SEILMAG_CSV,
		SEILMAG_CSV2,
		SEILMAG_CSV3,
		ULLERN_CSV,
		M2S_CSV,
		SAILWAVE_RACE_HTML,
		SAILWAVE_SUMMARY_HTML,
		SAILWAVE_SUMMARY2_HTML
	}

	private DataFileType dataFileType;
	private String dataFileName;

	public RaceData(String dataFileName, DataFileType dataFileType) {
		this.dataFileName = dataFileName;
		this.dataFileType = dataFileType;
	}

	public DataFileType getDataFileType() {
		return dataFileType;
	}

	public void setDataFileType(DataFileType dataFileType) {
		this.dataFileType = dataFileType;
	}

	public String getDataFileName() {
		return dataFileName;
	}

	public void setDataFileName(String dataFileName) {
		this.dataFileName = dataFileName;
	}

}

File: src/main/kotlin/no/omb/bigboat/data/RaceData.kt
package no.omb.bigboat.data

data class RaceData(
    val dataFileName: String,
    val dataFileType: DataFileType
) {
    enum class DataFileType {
        SEILMAG_CSV,
        SEILMAG_CSV2,
        SEILMAG_CSV3,
        SEILMAG_CSV4,
        ULLERN_CSV,
        DROBAK_CSV,
        M2S_CSV,
        M2S_CSV2,
        M2S_CSV3,
        M2S_CSV4,
        M2S_CSV6,
        SAILWAVE_RACE_HTML,
        SAILWAVE_SUMMARY_HTML,
        SAILWAVE_SUMMARY2_HTML
    }

    companion object {
        val races = arrayOf(
            RaceData("Bundefjorden Rundt", DataFileType.M2S_CSV4),
            RaceData("Bjørvikasprinten", DataFileType.M2S_CSV6),
            RaceData("Isbrytern", DataFileType.M2S_CSV),
            RaceData("Skagen Offshore Race", DataFileType.M2S_CSV6),
            RaceData("Oslofjorden Rundt", DataFileType.M2S_CSV3),
            RaceData("Drøbak Tristein", DataFileType.DROBAK_CSV),
            RaceData("Færderseilasen", DataFileType.M2S_CSV4),
            //RaceData("Grillcup", DataFileType.M2S_CSV3),
            //RaceData("Håøya Rundt", DataFileType.M2S_CSV3),
            RaceData("Asker Rundt", DataFileType.M2S_CSV2),
            RaceData("Hollænderseilasen", DataFileType.M2S_CSV3),
            //RaceData("Øyseilasen", DataFileType.M2S_CSV2),
            //RaceData("SætreKruset", DataFileType.SEILMAG_CSV4),
            //RaceData("Nesodden Høstcup", DataFileType.M2S_CSV4)
        )
    }
}
